tests/unit/amo/components/TestRatingsByStar.js

```diff
@@ -213,34 +227,65 @@ describe(__filename, () => {
       1: 0,
     };
     const addon = addonForGrouping(grouping);
+    const clientApp = CLIENT_APP_FIREFOX;
+    const lang = 'en-US';
+
+    dispatchClientMetadata({ clientApp, lang, store });
     store.dispatch(setGroupedRatings({ addonId: addon.id, grouping }));
+
     const root = render({
       addon,
+      history: fakeHistory,
       location: createFakeLocation({ query: { src } }),
+      store,
     });
-    const counts = root.find('.RatingsByStar-count').find(Link);
-
-    function validateLink(link, score) {
-      expect(link).toHaveProp(
-        'to',
-        reviewListURL({ addonSlug: addon.slug, score, src }),
+    const tableRow = root.find('.RatingByStar-table-row');
+
+    function validateLink(row, score) {
+      row.simulate('click', createFakeEvent());
+
+      sinon.assert.calledWith(
+        fakeHistory.push,
+        `/${lang}/${clientApp}${reviewListURL({
+          addonSlug: addon.slug,
+          score,
+          src,
+        })}`,
       );
     }
 
-    validateLink(counts.at(0), '5');
-    validateLink(counts.at(1), '4');
-    validateLink(counts.at(2), '3');
-    validateLink(counts.at(3), '2');
-    validateLink(counts.at(4), '1');
+    validateLink(tableRow.at(0), '5');
+    validateLink(tableRow.at(1), '4');
+    validateLink(tableRow.at(2), '3');
+    validateLink(tableRow.at(3), '2');
+    validateLink(tableRow.at(4), '1');
   });
 
   it('renders IconStar', () => {
     const root = render();
     const star = root.find(IconStar).at(0);
+
     expect(star).toHaveLength(1);
     expect(star).toHaveProp('selected');
   });
 
+  it('renders when star count is undefined', () => {
+    let fiveStarCount;
+    const grouping = {
+      5: fiveStarCount,
```

Comment:
Do you need to create this `fiveStarCount` variable, or could you just have `5: undefined,` here?

Reply:
Yes, we can directly write undefined but I just thought the case when `undefined` is possible, it happens only when variable is not initialized. But, yes it not necessary. I will update that.